libs/testdiff/testdiff.go

```diff
@@ -0,0 +1,91 @@
+package testdiff
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/stretchr/testify/assert"
+	"github.com/wI2L/jsondiff"
+)
+
+func UnifiedDiff(filename1, filename2, s1, s2 string) string {
+	edits := myers.ComputeEdits(span.URIFromPath(filename1), s1, s2)
+	return fmt.Sprint(gotextdiff.ToUnified(filename1, filename2, s1, edits))
+}
+
+func AssertEqualTexts(t testutil.TestingT, filename1, filename2, expected, out string) {
+	if len(out) < 1000 && len(expected) < 1000 {
+		// This shows full strings + diff which could be useful when debugging newlines
+		assert.Equal(t, expected, out)
+	} else {
+		// only show diff for large texts
+		diff := UnifiedDiff(filename1, filename2, expected, out)
+		t.Errorf("Diff:\n" + diff)
+	}
+}
+
+func AssertEqualJQ(t *testing.T, expectedName, outName, expected, out string, ignorePaths []string) {
```

Comment:
Use `testutil.TestingT` for compat with wrappers such as `acc.WorkspaceTest`.

Reply:
I don't understand, what is not compatible right now? 

BTW, I'm trying not to add dependency from libs on packages outside of libs, such as internal/testutil/

To be discussed, but I think we should get rid of all existing cases like that and enforce that.

Reply:
The `acc.WorkspaceTest` helper returns a wrapper type of `testing.T`. That means you cannot use that return value as argument here. The above function does take the `testutil.TestingT` so this one can too.

Separately, we can move testutil to libs as well to avoid deps on internal, but practically it doesn't make much of a difference IMO.

Reply:
updated.